types/mongodb/index.d.ts

```diff
@@ -38,41 +38,54 @@ export class MongoError extends Error {
     code?: number;
 }
 
-//http://mongodb.github.io/node-mongodb-native/2.1/api/MongoClient.html#.connect
-export interface MongoClientOptions {
-    uri_decode_auth?: boolean;
-    db?: DbCreateOptions;
-    server?: ServerOptions;
-    replSet?: ReplSetOptions;
-    mongos?: MongosOptions;
-    promiseLibrary?: Object;
-}
-
-// See : http://mongodb.github.io/node-mongodb-native/2.1/api/Db.html
+//http://mongodb.github.io/node-mongodb-native/2.2/api/MongoClient.html#.connect
+export interface MongoClientOptions extends
+    DbCreateOptions ,
+    ServerOptions   ,
+    MongosOptions   ,
+    ReplSetOptions  ,
+    SocketOptions {
+        loggerLevel         ?: string       ;// The logging level (error/warn/info/debug)
+        logger              ?: Object       ;// Custom logger object
+        
+        servername          ?: string       ;// String containing the server name requested via TLS SNI.
+        ha                  ?: boolean      ;// Default: true; Turn on high availability monitoring.
+        haInterval          ?: number       ;// Default: 10000; The High availability period for replicaset inquiry
+        domainsEnabled      ?: boolean      ;// Default: false;
+        
+        poolSize            ?: number       ;// Default:5; Number of connections for each server instance
+        // SSL configuration
+        ssl                 ?: boolean                  ;// Use ssl connection (needs to have a mongod server with ssl support)
+        sslValidate         ?: Object                   ;// Default: true; Validate mongod server certificate against ca (mongod server >=2.4 with ssl support required)
+        checkServerIdentity ?: boolean | Function       ;// Default: true; Server identity checking during SSL
+        sslCA               ?: Array<Buffer | string>   ;// Array of valid certificates either as Buffers or Strings
+        sslCRL              ?: Buffer                   ;// SSL Certificate revocation list binary buffer
+        sslCert             ?: Buffer | string          ;// SSL Certificate binary buffer
+        sslKey              ?: Buffer | string          ;// SSL Key file binary buffer
+        sslPass             ?: Buffer | string          ;// SSL Certificate pass phrase
+    
+        validateOptions     ?: Object                   ;// Default: false;
+        promiseLibrary      ?: Object                   ;// ES6 compatible promise constructor
+
+}
+// http://mongodb.github.io/node-mongodb-native/2.2/api/Db.html
 export interface DbCreateOptions {
```

Comment:
I meant the spacing between the property name and the type and the type and the `;`